### ISSUE TITLE
[Support]: Cannot save clips from HA app or web browser

### ISSUE THREAD

**Issue by LaurenceGough** [blakeblackshear/frigate#2992]
### Describe the problem you are having

Hello,

First of all thank you for this amazing project. I had a bit of a rough time getting hardware acceleration to work correctly on my Pi 4 but the latest Frigate BETA 0.11.0 (Full access) on my Home assistant OS install fixed that problem perfectly.

An issue I am seeing now - and I cannot currently confirm if this only impacts the BETA release is that I cannot save clips to my phone or PC when using the "Download Clip" option. It states the following when attempting to download on Chrome:

"Failed - Unknown server error. Please try again, or contact the server administrator."
If I click on the file that Chrome tried to download it just says the following on the webpage: 
401: Unauthorized
The URL of this download attempt page is: https://Home assistant URL here: /api/hassio_ingress/randomstringhere/api/events/randomstringhere-wioaj5 clip.mp4?download=true

I am able to watch the clips in Home Assistant OS just fine.

When using the official HA app it just does nothing, no errors etc are shown.

Sorry if this is a known issue.

Thank you,

Laurence

### Version

0.11.0-CBF26E0

### Frigate config file

```yaml
mqtt:
  host: x
  user: x
  password: x

detectors:
  cpu1:
    type: cpu
    num_threads: 4
  cpu2:
    type: cpu
    num_threads: 4

ffmpeg:
  hwaccel_args:
    - -c:v
    - h264_v4l2m2m

record:
      enabled: True

cameras:
  x:
    ffmpeg:
      inputs:
        - path: rtsp://xxxxxxxxxx@xxxxxxxxxx/Streaming/Channels/102
          roles:
            - detect
        - path: rtsp://xxxxxxxx@xxxxxxxxxxx/Streaming/Channels/101
          roles:
            - record
            - rtmp
    rtmp:
      enabled: True
    detect:
        width: 640
        height: 360
    motion:
        mask:
            - x
            - x
    zones:
        xzone:
            coordinates: x

  x:
    ffmpeg:
      inputs:
        - path: rtsp://x@x/live/ch1
          roles:
            - detect
        - path: rtsp://x@x/live/ch0
          roles:
            - record
            - rtmp
    rtmp:
      enabled: True
    detect:
        width: 768
        height: 432
        fps: 5
    motion:
        mask:
            - x

birdseye:
  enabled: True
  width: 1280
  height: 720
  quality: 8
  mode: continuous
```

### Relevant log output

```shell

From Frigate addon logs:

[2022-03-19 23:26:47] frigate.http                   ERROR   : ffmpeg version 4.3.3-0+rpt2+deb11u1 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr --extra-version=0+rpt2+deb11u1 --toolchain=hardened --incdir=/usr/include/aarch64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-mmal --enable-neon --enable-v4l2-request --enable-libudev --enable-epoxy --enable-sand --libdir=/usr/lib/aarch64-linux-gnu --arch=arm64 --enable-pocketsphinx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x5569bd4ff0] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x5569bd4ff0] moov atom not found
[concat @ 0x5569bd3fd0] Impossible to open 'pipe:/media/frigate/recordings/2022-03/19/20/camname1here/41.09.mp4'
pipe:: Invalid data found when processing input
[2022-03-19 23:26:51] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:57202]
[2022-03-19 23:26:58] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:57304]
[2022-03-19 23:27:04] ws4py                          INFO    : Terminating websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:57304]
[2022-03-19 23:27:21] ws4py                          INFO    : Managing websocket [Local => 127.0.0.1:5002 | Remote => 127.0.0.1:57350]
[2022-03-19 23:27:28] frigate.http                   ERROR   : ffmpeg version 4.3.3-0+rpt2+deb11u1 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 10 (Debian 10.2.1-6)
  configuration: --prefix=/usr --extra-version=0+rpt2+deb11u1 --toolchain=hardened --incdir=/usr/include/aarch64-linux-gnu --enable-gpl --disable-stripping --enable-avresample --disable-filter=resample --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librsvg --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-mmal --enable-neon --enable-v4l2-request --enable-libudev --enable-epoxy --enable-sand --libdir=/usr/lib/aarch64-linux-gnu --arch=arm64 --enable-pocketsphinx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-shared
  libavutil      56. 51.100 / 56. 51.100
  libavcodec     58. 91.100 / 58. 91.100
  libavformat    58. 45.100 / 58. 45.100
  libavdevice    58. 10.100 / 58. 10.100
  libavfilter     7. 85.100 /  7. 85.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  7.100 /  5.  7.100
  libswresample   3.  7.100 /  3.  7.100
  libpostproc    55.  7.100 / 55.  7.100
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x557a6f7f90] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x557a6f7f90] moov atom not found
[concat @ 0x557a6f6fd0] Impossible to open 'pipe:/media/frigate/recordings/2022-03/19/16/camname2here/17.22.mp4'
pipe:: Invalid data found when processing input



"Failed - Unknown server error. Please try again, or contact the server administrator."
If I click on the file that Chrome tried to download it just says the following on the webpage: 
401: Unauthorized
The URL of this download attempt page is: https://Home assistant URL here: /api/hassio_ingress/randomstringhere/api/events/randomstringhere-wioaj5 clip.mp4?download=true

When using the official HA app it just does nothing, no errors etc are shown.

```

### FFprobe output from your camera

```shell
Not sure if this is relevant - happens on all cams
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

N/A for this issue I believe

**Comment by NickM-27** [comment]
A few things to note here:

1. 0.11.0 isn't a beta yet and hasn't been released so that's not necessarily recommended. 
2. Are your cameras encoded in H.265 or H.264?
3. Have you confirmed that the recording segments are populated correctly? Does playing the clips work?

**Comment by LaurenceGough** [comment]
> 

Hi Nick, wow thanks for the super speedy reply!

The reason I had to use that version was due to the hardware decoding issue I was having with at least one camera generally being green. In the notes for this release this was mentioned as resolved. My poor little Pi 4 is trying its best without a Coral USB accelerator which I, like I'm sure many have on backorder!! I was just trying to give it some chance, and to be fair it's doing a good job now I got it a large heatsink case.

They are both H.264 streaming - no H.265.

Sorry I forgot to say, yes the clips playback just fine so clearly it has a copy it can play - unfortunately I cannot find a workaround way to download them, although I'm sure if I was better at the shell I could find them.

Thank you :).

**Comment by LaurenceGough** [comment]
I have just noticed it logs some entries in Frigate Addon logs. I have edited my post above to include this but here is a snippet below for reference, this issue happens with both of my cameras - both of different makes (one is a Hikvision).

"[mov,mp4,m4a,3gp,3g2,mj2 @ 0x557a6f7f90] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
[mov,mp4,m4a,3gp,3g2,mj2 @ 0x557a6f7f90] moov atom not found
[concat @ 0x557a6f6fd0] Impossible to open 'pipe:/media/frigate/recordings/2022-03/19/16/camname2here/17.22.mp4'
pipe:: Invalid data found when processing input"

**Comment by NickM-27** [comment]
Oh okay, didn't realize you were using the special 0.11.0 for the Pi. 

Have you tried accessing home assistant on the local IP and seeing if it works there?

**Comment by LaurenceGough** [comment]
> Oh okay, didn't realize you were using the special 0.11.0 for the Pi.
> 
> Have you tried accessing home assistant on the local IP and seeing if it works there?

No problem and great idea! I was hopeful that would work but sadly same issue - the download URL does update with the IP address but still the same problem sadly. :(

**Comment by blakeblackshear** [comment]
This is a known issue in the 0.11.0 beta. It has been fixed already, and the fix will be in the first official beta for 0.11.0. Not sure when that will be ready.

**Comment by LaurenceGough** [comment]
> This is a known issue in the 0.11.0 beta. It has been fixed already, and the fix will be in the first official beta for 0.11.0. Not sure when that will be ready.

Ah thanks Blake :). I'm looking forward to it, I am happy to help test anything at all. 
Enjoy the rest of the weekend guys!

**Comment by LaurenceGough** [comment]
Known issue as above so closing. Thank you.
